orchestrator-bundle/orc8r-bootstrapper-operator/src/charm.py

```diff
@@ -107,42 +100,98 @@ def _service_is_running(self) -> bool:
+            return None
+        return replicas.data[self.app].get("bootstrapper_private_key", None)
 
     @property
-    def _cert_bootstrapper_relation_created(self) -> bool:
-        """Returns whether cert-bootstrapper relation is created.
+    def _bootstrapper_private_key_is_stored(self) -> bool:
+        """Returns whether certificates are stored in relation data.
 
         Returns:
-            bool: True/False
+            bool: Whether certificates have been generated.
         """
-        return self._relation_created("cert-bootstrapper")
-
-    def _relation_created(self, relation_name: str) -> bool:
-        """Returns whether given relation was created.
+        if not self._bootstrapper_private_key:
+            logger.info("Bootstrapper private key not stored")
+            return False
+        return True
 
-        Args:
-            relation_name (str): Relation name
+    @property
+    def _replicas_relation_created(self) -> bool:
+        """Returns whether the replicas Juju relation was crated.
 
         Returns:
-            bool: True/False
+            str: Whether the relation was created.
         """
-        try:
-            if self.model.get_relation(relation_name):
-                return True
+        if not self.model.get_relation("replicas"):
             return False
-        except KeyError:
+        return True
+
+    @property
+    def _bootstrapper_private_key_is_pushed(self) -> bool:
+        """Returns whether bootstrapper private key is pushed to workload.
+
+        Returns:
+            bool: Whether bootstrapper private key is pushed to workload.
+        """
+        if not self._container.exists(f"{self.CERTIFICATE_DIRECTORY}/bootstrapper.key"):
+            logger.info("Bootstrapper private key is not pushed")
             return False
+        return True
+
+    def _on_install(self, event: InstallEvent) -> None:
+        """Triggered on charm install.
+
+        Args:
+            event: Juju event
 
-    def _on_magma_orc8r_bootstrapper_pebble_ready(
-        self, event: Union[PebbleReadyEvent, PrivateKeyAvailableEvent]
-    ) -> None:
+        Returns:
+            None
+        """
+        if not self._replicas_relation_created:
+            self.unit.status = WaitingStatus("Waiting for replicas relation to be created")
+            event.defer()
+            return
+        if not self._container.can_connect():
+            self.unit.status = WaitingStatus("Waiting for container to be ready")
+            event.defer()
+            return
+        if self.unit.is_leader():
+            bootstrapper_key = generate_private_key()
+            self._store_bootstrapper_private_key(bootstrapper_key.decode())
+        else:
+            if not self._bootstrapper_private_key_is_stored:
+                self.unit.status = WaitingStatus(
+                    "Waiting for leader to generate bootstrapper private key"
+                )
+                event.defer()
+                return
```

Comment:
A little bit less nesting?
```suggestion
        elif not self._bootstrapper_private_key_is_stored:
+            self.unit.status = WaitingStatus(
+                "Waiting for leader to generate bootstrapper private key"
+            )
+            event.defer()
+            return
```

Reply:
Done